google/cloud/storage/tests/async_client_integration_test.cc

```diff
@@ -178,6 +180,161 @@ TEST_F(AsyncClientIntegrationTest, StreamingRead) {
+          .get();
+  ASSERT_STATUS_OK(w);
+  AsyncWriter writer;
+  AsyncToken token;
+  std::tie(writer, token) = *std::move(w);
+
+  auto metadata = writer.Finalize(std::move(token)).get();
+  ASSERT_STATUS_OK(metadata);
+  ScheduleForDelete(*metadata);
+
+  EXPECT_EQ(metadata->bucket(), bucket_name());
+  EXPECT_EQ(metadata->name(), object_name);
+  EXPECT_EQ(metadata->size(), 0);
+}
+
+TEST_F(AsyncClientIntegrationTest, StreamingWriteMultiple) {
+  if (UsingEmulator()) GTEST_SKIP();
+  auto object_name = MakeRandomObjectName();
+  // Create a small block to send over and over.
+  auto constexpr kBlockSize = 256 * 1024;
+  auto constexpr kBlockCount = 16;
+  auto const block = MakeRandomData(kBlockSize);
+
+  auto client = AsyncClient();
+  auto w =
+      client.WriteObject(bucket_name(), object_name, gcs::IfGenerationMatch(0))
+          .get();
+  ASSERT_STATUS_OK(w);
+  AsyncWriter writer;
+  AsyncToken token;
+  std::tie(writer, token) = *std::move(w);
+  for (int i = 0; i != kBlockCount; ++i) {
+    auto p = writer.Write(std::move(token), WritePayload(block)).get();
+    ASSERT_STATUS_OK(p);
+    token = *std::move(p);
+  }
+
+  auto metadata = writer.Finalize(std::move(token)).get();
+  ASSERT_STATUS_OK(metadata);
+  ScheduleForDelete(*metadata);
+
+  EXPECT_EQ(metadata->bucket(), bucket_name());
+  EXPECT_EQ(metadata->name(), object_name);
+  EXPECT_EQ(metadata->size(), kBlockCount * kBlockSize);
+}
+
+TEST_F(AsyncClientIntegrationTest, StreamingWriteResume) {
+  if (UsingEmulator()) GTEST_SKIP();
+  auto object_name = MakeRandomObjectName();
+  // Create a small block to send over and over.
+  auto constexpr kBlockSize = 256 * 1024;
+  auto constexpr kInitialBlockCount = 4;
+  auto constexpr kTotalBlockCount = 4 + kInitialBlockCount;
+  auto constexpr kDesiredSize = kBlockSize * kTotalBlockCount;
+  auto const block = MakeRandomData(kBlockSize);
+
+  auto client = AsyncClient();
+  auto w =
+      client.WriteObject(bucket_name(), object_name, gcs::IfGenerationMatch(0))
+          .get();
+  ASSERT_STATUS_OK(w);
+  AsyncWriter writer;
+  AsyncToken token;
+  std::tie(writer, token) = *std::move(w);
+
+  auto const upload_id = writer.UploadId();
+  for (int i = 0; i != kInitialBlockCount - 1; ++i) {
+    auto p = writer.Write(std::move(token), WritePayload(block)).get();
+    ASSERT_STATUS_OK(p);
+    token = *std::move(p);
+  }
+
+  // Reset the existing writer and resume the upload.
+  writer = AsyncWriter();
+  w = client
+          .WriteObject(bucket_name(), object_name,
+                       gcs::UseResumableUploadSession(upload_id))
+          .get();
+  ASSERT_STATUS_OK(w);
+  std::tie(writer, token) = *std::move(w);
+  ASSERT_EQ(writer.UploadId(), upload_id);
+  auto const persisted = writer.PersistedState();
+  ASSERT_THAT(persisted, VariantWith<std::int64_t>(Le(kDesiredSize)));
+  auto offset = static_cast<int>(absl::get<std::int64_t>(persisted));
```

Comment:
Q.  Why drop to `int`?

Reply:
Because otherwise I had to cast a bunch of expressions from `std::int64_t` to `int` (or `std::size_t` or 32-bit platforms). Let me add a comment.